share/spice/search_code/search_code.js

```diff
@@ -1,52 +1,67 @@
-function ddg_spice_search_code(api_response) {
+(function(env) {
     "use strict";
+    env.ddg_spice_search_code = function(api_result) {
 
-    var query = api_response.query;
-    var data = api_response.results;
+        var query = api_result.query;
+        var data = api_result.results;
 
-    if(!data.length || data.length === 0) {
-        return Spice.failed('search_code');
-    }
-
-    var result;
-    for (var i = 0; i < data.length; i++) {
-        var checkRelevancy = [data[i].name, data[i].displayname, data[i].namespace].join(" ");
-        if (DDG.isRelevant(checkRelevancy, [], 2)) {
-            result = data[i];
-            break;
+        if(!data.length || data.length === 0) {
+            return Spice.failed('search_code');
         }
-    }
 
-    if (!result) {
-        return Spice.failed('search_code');
-    }
+        var result;
+        for (var i = 0; i < data.length; i++) {
+            var checkRelevancy = [data[i].name, data[i].displayname, data[i].namespace].join(" ");
+            if (DDG.isRelevant(checkRelevancy, [], 2)) {
+                result = data[i];
+                break;
+            }
+        }
 
-    function formatName(result) {
-        var formatted_name = result.name;
+        if (!result) {
+            return Spice.failed('search_code');
+        }
 
-        if (result.displayname !== '' || result.namespace !== '') {
-            formatted_name += ' (';
-            if (result.displayname !== '') {
-                formatted_name += result.displayname;
+        function formatTitle(result) {
+            var formatted_title = result.name;
+            if (result.displayname && result.displayname !== '') {
+                formatted_title += ' (' + result.displayname + ')';
             }
-            if (result.namespace !== '') {
-                formatted_name += (result.displayname ? ', ' : '') + result.namespace;
+            return formatted_title;
+        }
+
+        function formatSubtitle(result) {
+            var formatted_subtitle = null;
+            if (result.namespace && result.namespace !== '') {
+                formatted_subtitle = result.namespace;
             }
-            formatted_name += ')';
+            return formatted_subtitle;
         }
 
-        return formatted_name;
-    }
+        result.title = formatTitle(result);
+        result.subtitle = formatSubtitle(result);
 
-    Spice.add({
-        data             : result,
-        header1          : formatName(result),
-        sourceUrl       : 'https://searchcode.com/?q=' + query,
-        sourceName      : 'search[code]',
-        templates: {
-            item: Spice.search_code.search_code,
-            detail: Spice.search_code.search_code
-        },
-        
-    });
-}
+        Spice.add({
+            id: 'search_code',
+            name: 'Answer',
+            data: result,
+            meta: {
+                sourceURL: 'https://searchcode.com/?q=' + query,
```

Comment:
Fair enough. Stupid of me to not do this at first place. Thanks :)

On Thursday 16 June 2016, Zaahir Moolla <notifications@github.com> wrote:

> In share/spice/search_code/search_code.js
> <https://github.com/duckduckgo/zeroclickinfo-spice/pull/2810#discussion_r67369330>
> :
>
> > -        header1          : formatName(result),
> > -        sourceUrl       : 'https://searchcode.com/?q=' + query,
> > -        sourceName      : 'search[code]',
> > -        templates: {
> > -            item: Spice.search_code.search_code,
> > -            detail: Spice.search_code.search_code
> > -        },
> > -
> > -    });
> > -}
> > +        Spice.add({
> > +            id: 'search_code',
> > +            name: 'Answer',
> > +            data: result,
> > +            meta: {
> > +                sourceURL: 'https://searchcode.com/?q=' + query,
>
> Roughly something like...
>
> var subtitle_array = [ ];if (item.subtitle) {
>     subtitle_array.push(item.subtitle);
> }if ($reference) {
>     subtitle_array.push(item.subtitle);
> }
>
> subtitle: subtitle_array;
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/duckduckgo/zeroclickinfo-spice/pull/2810/files/13b3358c26c7b0472d5db0e4f6cd6c91d8ba2daa#r67369330>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AE9xBWV_Zw0hwkHF3M1S1ze8d4qpOzCcks5qMW7OgaJpZM4I0UML>
> .
>


-- 
Regards
*Sahil Dua*
Netaji Subhas Institute of Technology, New Delhi
Connect on LinkedIn <http://in.linkedin.com/in/sahildua2305>
*www.sahildua.com <http://www.sahildua.com>*